app/models/blacklight/primo_central/search_builder_behavior.rb

```diff
@@ -122,6 +122,10 @@ def process_date_range_query(primo_central_parameters)
 
       min = params.dig("range", "creationdate", "begin")
       max = params.dig("range", "creationdate", "end")
+      min = 1000 if min.to_s.match(/^\d+$/) && min.to_i < 1000
+      if max.to_s.match(/^\d+$/) and max.to_i < min.to_i
+        max = min
```

Comment:
```suggestion
        (min, max) = [max, min]
```

I think this needs to swap the values, otherwise you will have effectively set both to the same value always.

Also maybe a test case for this context ... i.e. check that values get swapped?